packages/react-widgets/src/widgets/utils/sortArrayByPropValues.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Sorts a list of objects based on a property values
+ * @param {object[]} array list of objects to sort
+ * @param {unknown[]} values values to sort by
+ * @param {string} prop property to sort by
+ * @returns sorted array
+ */
+export default function sortArrayByPropValues(array, values, prop) {
+  if (values?.length && prop) {
+    const uniqueValues = [...new Set(values)];
+    const elementsWithProp = array.filter(
+      (element) => element.hasOwnProperty(prop) && element[prop]
+    );
+    return [
+      ...array.filter((element) => !element.hasOwnProperty(prop)),
+      ...elementsWithProp.filter((item) => uniqueValues.indexOf(item[prop]) === -1),
+      ...elementsWithProp
+        .filter((item) => uniqueValues.indexOf(item[prop]) !== -1)
+        .sort((a, b) => uniqueValues.indexOf(b[prop]) - uniqueValues.indexOf(a[prop]))
+    ];
+  }
+
+  return array;
+}
```

Comment:
I don't think it's bad but I think that it's overengineering.

Reasons:
1. You create an utils to be used in only one place.
2. This util covers a lot of use cases that aren't necessary, having a certain impact in the performance.

Keep it simple, I think that this is far enough:
```js
Object.values(layers)
    .filter(({ legend }) => legend)
    .sort((layerA, layerB) => order.indexOf(layerB.id) - order.indexOf(layerA.id))
```

You can add there asserts to check that the id exists if you want, but I think it's also unnecessary because you cannot add layers without an id.